visual_regression_tracker/types.py

```diff
@@ -69,16 +80,13 @@ def __init__(self, test_run_response: TestRunResponse, api_url: str):
 
 
 def _to_dict(obj):
-    fields = dataclasses.fields(obj)
-    data = {}
-    for field in fields:
-        value = getattr(obj, field.name)
-        if value is not None:
-            data[field.name] = value
+    def dict_factory(key_values):
+        return dict(kv for kv in key_values if kv[1] is not None)
+
+    data = dataclasses.asdict(obj, dict_factory=dict_factory)
     return data
 
 
 def _from_dict(data, clazz):
-    fields = dataclasses.fields(clazz)
-    obj = clazz(*[data.get(field.name, None) for field in fields])
+    obj = dacite.from_dict(clazz, data)
```

Comment:
seems like this line is the reason for wrong type casting 
not sure how to let the `str` to accept `None` as well that comes from server

Reply:
oh yep, i should have expected that. i can fix it.

Reply:
I've resolved the specific failures of your new tests, but I'm not certain that this covers all user cases. Do you mind taking a look and suggesting any extra fields that need to be optional?

Reply:
sure, gonna check it!